src/contracts/pods/EigenPod.sol

```diff
@@ -164,267 +175,355 @@ contract EigenPod is IEigenPod, Initializable, ReentrancyGuardUpgradeable, Eigen
+            if (Endian.fromLittleEndianUint64(validatorFields[BeaconChainProofs.VALIDATOR_WITHDRAWABLE_EPOCH_INDEX]) <= slot/BeaconChainProofs.SLOTS_PER_EPOCH) {
+                _processFullWithdrawal(withdrawalAmountGwei, validatorIndex, validatorPubkeyHash, beaconChainETHStrategyIndex, podOwner, _validatorPubkeyHashToInfo[validatorPubkeyHash].status, slot);
+            } else {
+                _processPartialWithdrawal(slot, withdrawalAmountGwei, validatorIndex, validatorPubkeyHash, podOwner);
+            }
         }
     }
 
     function _processFullWithdrawal(
         uint64 withdrawalAmountGwei,
         uint40 validatorIndex,
+        bytes32 validatorPubkeyHash,
         uint256 beaconChainETHStrategyIndex,
         address recipient,
-        VALIDATOR_STATUS status
+        VALIDATOR_STATUS status,
+        uint64 withdrawalHappenedSlot
     ) internal {
         uint256 amountToSend;
+        uint256 withdrawalAmountWei;
 
-        // if the validator has not previously been proven to be "overcommitted"
-        if (status == VALIDATOR_STATUS.ACTIVE) {
-            // if the withdrawal amount is greater than the REQUIRED_BALANCE_GWEI (i.e. the amount restaked on EigenLayer, per ETH validator)
-            if (withdrawalAmountGwei >= REQUIRED_BALANCE_GWEI) {
+        uint256 currentValidatorRestakedBalanceWei = _validatorPubkeyHashToInfo[validatorPubkeyHash].restakedBalanceGwei * GWEI_TO_WEI;
+        
+        /**
+        * If the validator is already withdrawn and additional deposits are made, they will be automatically withdrawn
+        * in the beacon chain as a full withdrawal.  Thus such a validator can prove another full withdrawal, and 
+        * withdraw that ETH via the queuedWithdrawal flow in the strategy manager. 
+        */
+        if (status == VALIDATOR_STATUS.ACTIVE || status == VALIDATOR_STATUS.WITHDRAWN) {
+            // if the withdrawal amount is greater than the MAX_VALIDATOR_BALANCE_GWEI (i.e. the amount restaked on EigenLayer, per ETH validator)
+            if (withdrawalAmountGwei > MAX_VALIDATOR_BALANCE_GWEI) {
                 // then the excess is immediately withdrawable
-                amountToSend = uint256(withdrawalAmountGwei - REQUIRED_BALANCE_GWEI) * uint256(GWEI_TO_WEI);
-                // and the extra execution layer ETH in the contract is REQUIRED_BALANCE_GWEI, which must be withdrawn through EigenLayer's normal withdrawal process
-                restakedExecutionLayerGwei += REQUIRED_BALANCE_GWEI;
+                amountToSend = uint256(withdrawalAmountGwei - MAX_VALIDATOR_BALANCE_GWEI) * uint256(GWEI_TO_WEI);
+                // and the extra execution layer ETH in the contract is MAX_VALIDATOR_BALANCE_GWEI, which must be withdrawn through EigenLayer's normal withdrawal process
+                withdrawableRestakedExecutionLayerGwei += MAX_VALIDATOR_BALANCE_GWEI;
+                withdrawalAmountWei = _calculateRestakedBalanceGwei(MAX_VALIDATOR_BALANCE_GWEI) * GWEI_TO_WEI;
```

Comment:
something seems off here!!
shouldn't `withdrawableRestakedExecutionLayerGwei` increment by `withdrawalAmountWei` ?

To illustrate what I think the BUG is:
What if `withdrawalAmountGwei = MAX_VALIDATOR_BALANCE_GWEI + 1`? With the existing code, won't we end up with `withdrawalAmountWei < MAX_VALIDATOR_BALANCE_GWEI`, meaning we will increase `withdrawableRestakedExecutionLayerGwei` by more than the `withdrawalAmountWei` ?

Reply:
I agree, withdrawableRestakedExecutionLayerGwei currently slightly overestimates - wouldn't be an actual issue because SM still controls exactly how much is decremented.

I think the fix is to directly withdraw any balance more than `_calculateRestakedBalanceGwei(MAX_VALIDATOR_BALANCE_GWEI)`